src/pages/lessons/[slug].tsx

```diff
@@ -79,14 +79,11 @@ const lessonLoader = (slug: string, token: string) => {
 
 const NextResourceButton: FunctionComponent<{
   path: string
-  onClick: () => void
   className: string
-}> = ({children, path, onClick, className = ''}) => {
+}> = ({children, path, className = ''}) => {
   return (
     <Link href={path || '#'}>
-      <a className={className} onClick={onClick}>
-        {children || 'Next Lesson'}
-      </a>
+      <a className={className}>{children || 'Next Lesson'}</a>
```

Comment:
this was another case where `onclick` is very demanding, I went for another option instead and monitor for a route change to set the state.